st2common/tests/unit/services/test_workflow_service_retries.py

```diff
@@ -195,7 +196,6 @@ def test_retries_exhausted_from_coordinator_connection_error(self, mock_get_lock
         "update_task_state",
         mock.MagicMock(
             side_effect=[
-                mongoengine.connection.ConnectionFailure(),
```

Comment:
Without this, the side effect happens too much, resulting in this traceback + test error on at least python3.9 (and probably newer):
```
======================================================================
1) ERROR: test_recover_from_database_connection_error (tests.unit.services.test_workflow_service_retries.OrquestaServiceRetryTest)
----------------------------------------------------------------------
   Traceback (most recent call last):
    virtualenv/lib/python3.9/site-packages/mock/mock.py line 1452 in patched
      return func(*newargs, **newkeywargs)
    st2common/tests/unit/services/test_workflow_service_retries.py line 222 in test_recover_from_database_connection_error
      wf_svc.handle_action_execution_completion(tk1_ac_ex_db)
    virtualenv/lib/python3.9/site-packages/retrying.py line 56 in wrapped_f
      return Retrying(*dargs, **dkw).call(f, *args, **kw)
    virtualenv/lib/python3.9/site-packages/retrying.py line 266 in call
      raise attempt.get()
    virtualenv/lib/python3.9/site-packages/retrying.py line 301 in get
      six.reraise(self.value[0], self.value[1], self.value[2])
    virtualenv/lib/python3.9/site-packages/six.py line 719 in reraise
      raise value
    virtualenv/lib/python3.9/site-packages/retrying.py line 251 in call
      attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
    st2common/st2common/services/workflows.py line 969 in handle_action_execution_completion
      update_task_state(
    virtualenv/lib/python3.9/site-packages/mock/mock.py line 1178 in __call__
      return _mock_self._mock_call(*args, **kwargs)
    virtualenv/lib/python3.9/site-packages/mock/mock.py line 1182 in _mock_call
      return _mock_self._execute_mock_call(*args, **kwargs)
    virtualenv/lib/python3.9/site-packages/mock/mock.py line 1243 in _execute_mock_call
      raise result
   ConnectionFailure: 
```